Engine/src/Engine/Rendering/Window.cpp

```diff
@@ -74,30 +74,35 @@ namespace Engine
 		m_camera = newCamera;
 	}
 
+	//Adjust height to maintain current aspect ratio
 	void Window::resize(int width, int height)
 	{
-		//Adjust height to maintain current aspect ratio
-		int newHeight = height * getAspectRatio(); 
-		int letterboxHeight = height - newHeight;
-		glViewport(0, letterboxHeight / 2, width, newHeight);
+		m_windowWidth = width;
+		m_windowHeight = height;
+
 	}
 #pragma endregion
 
 #pragma region Getters
 	//Convert vec3 screenspace coordinates to a vec3 of the worldspace coordinates.
-	glm::vec3 Window::screenSpaceToWorldSpace(const glm::vec2& screenSpaceVector)
+	glm::vec3 Window::screenSpaceToWorldSpace(const glm::vec2& screenSpaceVector, glm::vec2 debugWindowOffset = glm::vec2(0))
 	{
+		auto modifiedVector = screenSpaceVector - debugWindowOffset; //Offset the result by the position of the debug window
 		float frustumWidth = m_camera.frustumWidth;
 		float aspectRatio = m_camera.aspectRatio;
-		return glm::vec3(screenSpaceVector.x / m_windowWidth * frustumWidth, screenSpaceVector.x / m_windowWidth * frustumWidth * aspectRatio, 1);
+		
+		auto result = glm::vec3((modifiedVector.x / m_windowWidth) * frustumWidth, (modifiedVector.y / m_windowHeight) * frustumWidth * aspectRatio, 1);
```

Comment:
Math

Reply:
Fixed, now uses width over height instead of height over width for the aspect ratio.